src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java

```diff
@@ -0,0 +1,517 @@
+{
+
+    /**
+     * Tools and Items needed by the worker.
+     */
+    public final List<ToolType>  toolsNeeded = new ArrayList<>();
+    public final List<ItemStack> itemsNeeded = new ArrayList<>();
+
+    /**
+     * Amount of animals needed to bread.
+     */
+    public static final int NUM_OF_ANIMALS_TO_BREED = 2;
+
+    /**
+     * Butchering attack damage.
+     */
+    private static final int BUTCHERING_ATTACK_DAMAGE = 5;
+
+    /**
+     * Distance two animals need to be inside to breed.
+     */
+    private static final int DISTANCE_TO_BREED = 10;
+
+    /**
+     * Delays used to setDelay()
+     */
+    public static final int BUTCHER_DELAY    = 20;
+    public static final int DECIDING_DELAY   = 40;
+    public static final int BREEDING_DELAY   = 40;
+    public static final int NO_ANIMALS_DELAY = 100;
+
+    /**
+     * Number of actions needed to dump inventory.
+     */
+    private static final int ACTIONS_FOR_DUMP = 10;
+
+    /**
+     * Creates the abstract part of the AI.
+     * Always use this constructor!
+     *
+     * @param job the job to fulfill.
+     */
+    public AbstractEntityAIHerder(@NotNull final J job)
+    {
+        super(job);
+        super.registerTargets(
+          new AITarget(IDLE, START_WORKING),
+          new AITarget(START_WORKING, this::startWorkingAtOwnBuilding),
+          new AITarget(PREPARING, this::prepareForHerding),
+          new AITarget(HERDER_DECIDE, this::decideWhatToDo),
+          new AITarget(HERDER_BREED, this::breedAnimals),
+          new AITarget(HERDER_BUTCHER, this::butcherAnimals),
+          new AITarget(HERDER_PICKUP, this::pickupItems)
+        );
+        worker.setCanPickUpLoot(true);
+    }
+
+    @Override
+    protected int getActionsDoneUntilDumping()
+    {
+        return ACTIONS_FOR_DUMP;
+    }
+
+    @NotNull
+    @Override
+    protected List<ItemStack> itemsNiceToHave()
+    {
+        final List<ItemStack> list = super.itemsNiceToHave();
+        list.add(getBreedingItems());
+        return list;
+    }
+
+    /**
+     * Decides what job the herder should switch to, breeding or Butchering.
+     *
+     * @return The next {@link AIState} the herder should switch to, after executing this method.
+     */
+    public AIState decideWhatToDo()
+    {
+        setDelay(DECIDING_DELAY);
+
+        final List<T> animals = new ArrayList<>(searchForAnimals());
+
+        if (animals.isEmpty())
+        {
+            setDelay(NO_ANIMALS_DELAY);
+            return HERDER_DECIDE;
+        }
+
+        worker.setLatestStatus(new TextComponentTranslation(TranslationConstants.COM_MINECOLONIES_COREMOD_STATUS_HERDER_DECIDING));
+
+        final int numOfBreedableAnimals = ((int) animals.stream().filter(animal -> animal.getGrowingAge() == 0).count());
+
+        final boolean hasBreedingItem =
+          InventoryUtils.hasItemInItemHandler(new InvWrapper(worker.getInventoryCitizen()), getBreedingItem().getItem(), getBreedingItem().getItemDamage());
```

Comment:
Nope. This still falls back on the Item instead of using the ItemStack. Use the predicate version of the same method with areItemsEqualIgnoreStackSize.